src/common/codefix-handler.js

```diff
@@ -70,10 +70,15 @@ async function readCodeChangeReport(s3Client, bucketName, reportKey, log) {
       return null;
     }
 
-    // If reportData is a plain string (not JSON), wrap it in a diff object
+    // If reportData is a plain string, try to parse it as JSON
     if (typeof reportData === 'string') {
-      log.info(`Report data is plain text, wrapping in diff object for key: ${reportKey}`);
-      return { diff: reportData };
+      try {
+        const parsedData = JSON.parse(reportData);
+        return parsedData;
```

Comment:
@dmaurya929 can the format be published somewhere (if not done already) to have consistency across opportunities.

Reply:
Updated the JS doc, and it is mentioned in wiki also. We need to check for a string here because when saving report.json, the contentType metadata must be set to application/json. I’ll address this in the next iteration.